### PR TITLE
feat(core): Display conditions in displayOptions (no-changelog)

### PR DESCRIPTION
**Pull request by michael-radency**
supported conditions: `eq, not, gt, lt, lte, gte, between, includes, startsWith, endsWith`
usage example:

```
{
	name: 'string1',
	displayName: 'string1',
	displayOptions: {
		show: {
			number1: [{ _cnd: { between: { from: 1, to: 3 } } }],
		},
	},
	type: 'string',
	default: 'default string1',
},
```

```
nodePropertiesArray: [
	{
		name: 'mode',
		displayName: 'mode',
		type: 'string',
		default: 'mode1',
	},
	{
		name: 'string1',
		displayName: 'string1',
		displayOptions: {
			show: {
				mode: [{ _cnd: { startsWith: 'mo' } }],
			},
		},
		type: 'string',
		default: 'default string1',
	},
],
```

### CODE REVIEW

**Review by maspio — DISMISSED** [review]

**Review by maspio — APPROVED** [review]

File: packages/nodes-base/nodes/BambooHr/v1/actions/employee/create/description.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const employeeCreateDescription: EmployeeProperties = [
 		},
 		default: '',
 	},
-	...createEmployeeSharedDescription(true),
+	...(createEmployeeSharedDescription(true) as EmployeeProperties),
```
Comment (netroy) [inline packages/nodes-base/nodes/BambooHr/v1/actions/employee/create/description.ts:47]: this change doesn't seem related to the PR.

**Review by netroy — COMMENTED** [review]
Is there a ticket for this? The code isn't very clear about what we are trying to fix here. Also, this seems to fix some issue in the public api, but no where in the title or description is the public-api mentioned. 
It would be helpful to add more context to this PR for anyone who might end up on it while debugging any future issues.

**Comment by michael-radency** [comment]
@netroy 
here is linear ticket https://linear.app/n8n/issue/NODE-965/display-conditions-in-displayoptions
goal is to have more control when to show/hide node's parameters, not just equality check

**Comment by janober** [comment]
Got released with [`n8n@1.27.2`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.27.2)
